src/dmd/expression.d

```diff
@@ -3177,15 +3177,13 @@ extern (C++) final class StructLiteralExp : Expression
              *   __sl%s%d, where %s is the struct name
              */
             const size_t len = 10;
-            char[len] buf = void;
+            char[len + 1] buf = void;
+            buf[len] = 0;
+            strcpy(buf.ptr, "__sl");
+            strncat(buf.ptr, sd.ident.toChars(), len - 4 - 1);
+            assert(buf[len] == 0);
```

Comment:
proof that you're not sure how strncpy works :-)

Reply:
er, I mean strncat. I don't remember how either of them work, and nobody else does, either.

Reply:
Looking at the code, this `buf[len]` is set to `0` before `strncat()`, so this check is just to ensure that we didn't overflow.

Reply:
I know. And sure as shootin', the person who wrote that was unsure of what strncat actually does :-) I even WROTE strncat, and I don't remember what it does. I have to look it up every time.

There's no need for such "insurance" if straightforward memcpy's are done.

The code screams "there's a bug in it". Even if there isn't one, every maintainer is going to have to check it again. This is any strncat must die. BTW, most of the time when I check a strncat, it's being used wrong.